### ISSUE TITLE
Copilot Became "Extension is not enabled" State After Using Some Times

### ISSUE THREAD

**Issue by owenzhao** [github/CopilotForXcode#24]
**Describe the bug**

After using Copilot for some time, the icon of Copilot changes to an info icon style and prompts with "Extension is not enabled. Enable GitHub Copilot under Xcode and then restart Xcode". At this point, you don’t need to follow the prompt; simply closing and reopening Xcode should be enough. This is because the setting allowing Xcode extensions to use Copilot hasn’t changed. 

I’m not sure exactly how to reproduce it, but I suspect it might also be a system-related issue such as the authorization expiring incorrectly after 24 hours upon opening Xcode.

**Versions**
- Copilot for Xcode: [e.g. 0.27.93]
- Xcode: [e.g. 16.2 beta 2 (16C5013f)]
- macOS: [e.g. 15.2 Beta (24C5073e)]

**Steps to reproduce**
1. Open Xcode
2. Wait until the issue shows. Maybe the next day it will show.

**Screenshots**

<img width="586" alt="Screenshot 2024-11-06 at 5 17 00 AM" src="https://github.com/user-attachments/assets/dbc7b1fa-9f38-4d7b-892a-dcdab31fdd71">

**Logs**

[github-copilot-for-xcode.log](https://github.com/user-attachments/files/17638985/github-copilot-for-xcode.log)

**Additional context**

**Comment by devm33** [comment]
Hi @owenzhao, thanks for trying out Copilot for Xcode and providing feedback!

The warning status is important since without the extension active/enabled in Xcode, the app is forced to use a fallback that does not work as well.

I would be curious to know if you're able to figure out what might be causing the extension to get inactivated in Xcode. If anything changes with the app binary (e.g. an app update) that causes Xcode to deactivate the extension. That's the only case I'm aware of.

**Comment by nickjbedford** [comment]
I am experiencing this error as well. MacBook M1 Pro 14" with Sonoma 15.1 using Xcode 16.1 and the latest GitHub Copilot extension. I can use it for a minute or so then it gives me the orange notification in Xcode that it's falling back. It says to reactivate the extension, but it's already activated.

Also the blue "Launching service app" keeps trying to launch constantly.

<img width="486" alt="Screenshot 2024-11-15 at 7 18 09 pm" src="https://github.com/user-attachments/assets/b2f91578-b785-431e-b5c6-6b9a27a15407">
<img width="553" alt="Screenshot 2024-11-15 at 7 19 20 pm" src="https://github.com/user-attachments/assets/af005b32-8005-4c6d-ad12-e23cfc495cc5">
<img width="294" alt="Screenshot 2024-11-15 at 7 20 04 pm" src="https://github.com/user-attachments/assets/b799a760-eb44-4923-9d87-8a55640d91ae">

**Comment by owenzhao** [comment]
After I disabled "Quit GitHub Copilot when Xcode App is closed" option, this issue no longer appears.

However, I have encountered another issue that the Menu Tray Icon turned into circle state and there was no suggestions for Xcode from Copilot. I quit Copilot and relaunch it, it works again.

**Comment by devm33** [comment]
Hi @owenzhao that's interesting that keeping the app open prevents Xcode from disabling it.

>  I have encountered another issue that the Menu Tray Icon turned into circle state

Does it get stuck in the loading state indefinitely until you restarted? The loading indicator should be improved on the latest version 0.29.0

**Comment by owenzhao** [comment]
> Hi @owenzhao that's interesting that keeping the app open prevents Xcode from disabling it.
> 
> > I have encountered another issue that the Menu Tray Icon turned into circle state
> 
> Does it get stuck in the loading state indefinitely until you restarted? The loading indicator should be improved on the latest version 0.29.0

No. It only circled a few seconds and the pilot image showed. However, there was no suggestion in Xcode at that time. Now I am with version 0.29.0 and will keep this update if this happens again.
